File: content/04/code/js/09-delete-unused-step-definition/test/person.test.js
const assert = require('assert')
const sinon = require('sinon')
const {Person} = require('../src/shouty')

describe('Person', () => {
  let network, networkStub
  beforeEach(() => {
    network = {
      subscribe() {},
      broadcast() {}
    }
    networkStub = sinon.spy(network)
  })

  it('it subscribes to the network', () => {
    const lucy = new Person(network)
    assert(networkStub.subscribe.calledOnce)
    assert.strictEqual(networkStub.subscribe.getCall(0).args[0], lucy)
  })

  it('broadcasts shouts to the network', () => {
    const message = "Free bagels!"
    const sean = new Person(network)
    sean.shout(message)
    assert(networkStub.broadcast.calledOnce)
    assert.strictEqual(networkStub.broadcast.getCall(0).args[0], message)
  })

  it('remembers messages heard', () => {
    const message = "Free bagels!"
    const lucy = new Person(network)
    lucy.hear(message)
    assert.deepStrictEqual(lucy.messagesHeard(), [message])
  })
})

File: content/bdd-with-cucumber/04/code/java/07--step07-network-as-an-instance-field/shouty/src/test/java/shouty/PersonTest.java
package shouty;

import org.junit.Test;

import static java.util.Arrays.asList;
import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

public class PersonTest {
    private final Network network = mock(Network.class);

    @Test
    public void it_subscribes_to_the_network() {
        Person lucy = new Person(network);
        verify(network).subscribe(lucy);
    }

    @Test
    public void broadcasts_shouts_to_the_network() {
        String message = "Free bagels!";
        Person sean = new Person(network);
        sean.shout(message);
        verify(network).broadcast(message);
    }

    @Test
    public void remembers_messages_heard() {
        String message = "Free bagels!";
        Person lucy = new Person(network);
        lucy.hear(message);
        assertEquals(asList(message), lucy.getMessagesHeard());
    }
}